test/errhandling/misc/forExprReturnsRecOrThrows.chpl

```diff
@@ -0,0 +1,67 @@
+module M {
+  config const testForExpr = false;
+
+  /*
+   forall and bracketed versions are blocked by other bugs. See below.
+
+  config const testForAllExpr = false;
+  config const testBracketForAllExpr = false;
+   */
+
+  var initCounter = 0;
```

Comment:
The non-atomic counter won't be reliable when you enable the parallel forall/`[]` loops.   You might consider addressing that now.

Reply:
Good point. Made that change